platforms/software/build-init/src/integTest/groovy/org/gradle/buildinit/projectspecs/internal/BuildInitPluginProjectSpecsIntegrationTest.groovy

```diff
@@ -0,0 +1,340 @@
+                `java-library`
+            }
+        """)
+
+        kotlinFile("new-project/subproject/build.gradle.kts",
+            """
+                plugins {
+                    `java-library`
+                }
+            """
+        )
+
+        when:
+        initSucceedsWithPluginSupplyingSpec("org.barfuin.gradle.taskinfo:2.2.0")
+
+        then:
+        assertResolvedPlugin("org.barfuin.gradle.taskinfo", "2.2.0")
+        // TODO: should appear exactly once, but no way to automatically verify this currently.  Looking at the output, it is true currently
+    }
+
+    @LeaksFileHandles
+    def "can generate custom project type using argument to init"() {
+        given:
+        publishTestPlugin()
+
+        when:
+        initSucceedsWithPluginSupplyingSpec("org.example.myplugin:1.0")
+
+        then:
+        assertResolvedPlugin("org.example.myplugin", "1.0")
+        outputDoesNotContain("MyPlugin applied.")
+        assertLoadedSpec("First Project Type")
+        assertLoadedSpec("Second Project Type")
+
+        // Note: If running in non-interactive mode, first template is automatically used
+        assertProjectFileGenerated("project.output", "MyGenerator created this First Project Type project.")
+        assertWrapperGenerated()
+    }
+
+    @LeaksFileHandles
+    def "can generate custom project type when specifying multiple custom plugins using arguments to init"() {
+        given:
+        publishTestPlugin()
+
+        when:
+        initSucceedsWithPluginSupplyingSpec("org.example.myplugin:1.0,org.barfuin.gradle.taskinfo:2.2.0")
+
+        then:
+        assertResolvedPlugin("org.example.myplugin", "1.0")
+        assertResolvedPlugin("org.barfuin.gradle.taskinfo", "2.2.0")
+        outputDoesNotContain("MyPlugin applied.")
+        assertLoadedSpec("First Project Type")
+        assertLoadedSpec("Second Project Type")
+
+        // Note: If running in non-interactive mode, first template is used
+        assertProjectFileGenerated("project.output", "MyGenerator created this First Project Type project.")
+        assertWrapperGenerated()
+    }
+
+    @LeaksFileHandles
+    def "can generate a custom project type non-interactively using --type if multiple types are available using argument to init"() {
+        given:
+        publishTestPlugin()
+
+        when:
+        initSucceedsWithPluginSupplyingSpec("org.example.myplugin:1.0", "second-project-type")
+
+        then:
+        assertResolvedPlugin("org.example.myplugin", "1.0")
+        outputDoesNotContain("MyPlugin applied.")
+        assertLoadedSpec("First Project Type")
+        assertLoadedSpec("Second Project Type")
+
+        // Note: If running in non-interactive mode, first template is used
+        assertProjectFileGenerated("project.output", "MyGenerator created this Second Project Type project.")
+        assertWrapperGenerated()
+    }
+
+    @Ignore // TODO: No JDK 21 on CI yet, remove this when JDK 21 is available
```

Comment:
❌ What? Java 21 is available for the build, you may need to request it with:
https://github.com/gradle/gradle/blob/e7e7ed8c5b28c7f0458d4cd3e05c82b12a38e327/testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java#L140

Reply:
OK, thanks, I will give this a try.